// rrc/lib/ConfigSwitch.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Route = require('react-router/Route');

var _Route2 = _interopRequireDefault(_Route);

var _matchRoutes2 = require('./helpers/matchRoutes');

var _matchRoutes3 = _interopRequireDefault(_matchRoutes2);

var _makeInjectable = require('./helpers/makeInjectable');

var _makeInjectable2 = _interopRequireDefault(_makeInjectable);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ConfigSwitch = function ConfigSwitch(_ref, _ref2) {
  var routes = _ref.routes,
      location = _ref.location;
  var router = _ref2.router;
  var parent = router.route;

  var currentLocation = location || parent.location;

  var _matchRoutes = (0, _matchRoutes3.default)(routes, currentLocation.pathname, parent),
      match = _matchRoutes.match,
      route = _matchRoutes.route;

  var routeProps = match ? (0, _makeInjectable2.default)(route) : route;
  return match ? _react2.default.createElement(_Route2.default, _extends({}, routeProps, { location: currentLocation, computedMatch: match })) : null;
};

ConfigSwitch.propTypes = {
  routes: _propTypes2.default.array.isRequired,
  location: _propTypes2.default.shape({
    pathname: _propTypes2.default.string.isRequired
  })
};

ConfigSwitch.contextTypes = {
  router: _propTypes2.default.shape({
    route: _propTypes2.default.shape({
      location: _propTypes2.default.object.isRequired
    }).isRequired
  }).isRequired
};

exports.default = ConfigSwitch;
